opengrok-indexer/src/main/java/org/opengrok/indexer/util/Progress.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * See LICENSE.txt included in this distribution for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at LICENSE.txt.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+/*
+ * Copyright (c) 2007, 2019, Oracle and/or its affiliates. All rights reserved.
+ * Portions Copyright (c) 2017-2018, Chris Fraire <cfraire@me.com>.
+ */
+
+package org.opengrok.indexer.util;
+
+import org.opengrok.indexer.configuration.RuntimeEnvironment;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class Progress implements AutoCloseable {
+    private Logger logger;
+    private long totalCount;
+    private String prefix;
+
+    private AtomicLong currentCount = new AtomicLong();
+    private Thread loggerThread = null;
+    private AtomicBoolean run = new AtomicBoolean();
```

Comment:
The usual signal for this type of worker thread coordination is `volatile Boolean` since it is updated by only one thread and read by the other and no further synchronization is necessary — since there’s no harm to loop an additional time

Reply:
okay, thanks. changed.